src/librustdoc/clean/types.rs

```diff
@@ -1153,23 +1146,17 @@ impl Attributes {
     }
 
     /// Return the doc-comments on this item, grouped by the module they came from.
-    ///
     /// The module can be different if this is a re-export with added documentation.
-    crate fn collapsed_doc_value_by_module_level(&self) -> FxHashMap<Option<DefId>, String> {
-        let mut ret = FxHashMap::default();
-        if self.doc_strings.len() == 0 {
-            return ret;
-        }
-        let last_index = self.doc_strings.len() - 1;
-
-        for (i, new_frag) in self.doc_strings.iter().enumerate() {
-            let out = ret.entry(new_frag.parent_module).or_default();
-            add_doc_fragment(out, new_frag);
-            if i == last_index {
-                out.pop();
-            }
+    ///
+    /// The last newline is not trimmed so the produced strings are reusable between
+    /// early and late doc link resolution regardless of their position.
+    crate fn prepare_to_doc_link_resolution(&self) -> FxHashMap<Option<DefId>, String> {
```

Comment:
at this point it may as well be in `collect_intra_doc_links.rs`; it's not reusable.

Reply:
It uses `add_doc_fragment` which is private, but I can make it more public if that's preferable.

Reply:
Left as is for now, because all other impls on `Attributes` are here, and because of `add_doc_fragment`.
Maybe later I'll eliminate it entirely by changing how `Attributes` are represented.